client/src/portal/api/mod.rs

```diff
@@ -279,11 +286,224 @@ impl TryFrom<&[u8]> for Keyring {
     }
 }
 
-#[cfg(test)]
+#[derive(Debug)]
```

Comment:
Can you instead move this to a new separate module? to keep the legacy bits separate from the reset.

Reply:
Moved to `legacy_keyring.rs`.